addons/io_scene_gltf2/__init__.py

```diff
@@ -214,10 +214,19 @@ def __init__(self):
         default=False
     )
 
-    export_materials: BoolProperty(
+    export_materials: EnumProperty(
         name='Materials',
-        description='Export materials',
-        default=True
+        items=(('EXPORT', 'Export',
+                'Export materials '),
+                ('PLACEHOLDER', 'Placeholder',
+                'Do not export materials, but keep placeholder primitives '),
+                ('NONE', 'No export',
+                'Do not export materials '),
+               ),
```

Comment:
Suggested naming:

- **All**
  - _Include all materials, whether used or unused_
- **Active** (default) 
  - _Include all materials used by at least one mesh_
- **Placeholder**
  - _Include only material names, without properties or textures_
- **None**
  - _Do not export materials_

The "All" option does not need to be included in this PR, but could be added in the future if we decide to support https://github.com/KhronosGroup/glTF-Blender-IO/issues/1081.

Reply:
Currently, "placeholder" doesn't export material names (and it can't where there is an empty slot). The only difference with "None" is that None merged all primitives into a single one, where "Placeholder" will keep all primitives created

But OK with the concept of a future "All" option for #1081

Reply:
Oops, I see. In that case... I think this is a useful feature to have, but I would not create a material dropdown for it. Or at least not yet, perhaps if we add the "All" option later. How about a checkbox in the geometry section instead? Perhaps:

- **Merge vertex groups** (default = true)
  - When enabled, vertex groups sharing a material are merged for more efficient export.

Or perhaps I'm still misunderstanding. But I do like using Blender's term "vertex groups" rather than glTF's term "primitives", which users may not know.

Reply:
Or maybe better:

- **Vertex groups** (default = FALSE)
  - When enabled, vertex groups within a mesh are preserved, whether or not they require separate materials

Reply:
This isn't related to vertex groups.

Reply:
Confirmed, this is not linked to Vertex Group.

Reply:
Ok, I see. In that case, I can't tell from the description included for this UI option what the option does. Could you suggest a description that would help users to understand when to choose this?